src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java

```diff
@@ -414,4 +423,37 @@ private static synchronized void deleteEventToJenkinsJobRun(
         // clean up
         innerDeleteEventToJenkinsJobRun(build);
     }
+
+  /**
+   * Reconciles Jenkins job runs and OpenShift builds
+   *
+   * Deletes all job runs that do not have an associated build in OpenShift
+   */
+  private static synchronized void reconcileRunsAndBuilds() {
```

Comment:
is there any risk of this running after someone clicks "run build" in jenkins and before the build object is created in openshift?  (at which point the jenkins run would be deleted).

Reply:
there should not be a risk ... before a job run would end up in the `job.getBuilds` list, our BuildDecisionHandler plug point into jenkins gets called (the shouldSchedule method) ... shouldSchedule checks to see if the build is there before saying we should move forward

Reply:
good thought though

Reply:
cool